### PR TITLE
Lambda handler for removing an account

### PR DESCRIPTION
**Pull request by cj1993**
## What?

New handler for for removing an account (user profile + user credentials)

## Why?

Users want to be able to remove their account.

### CODE REVIEW

File: account-management-api/src/main/java/uk/gov/di/accountmanagement/lambda/RemoveAccountHandler.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+public class RemoveAccountHandler
+        implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+
+    private final AuthenticationService authenticationService;
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    public RemoveAccountHandler(
+            AuthenticationService authenticationService,
+            ValidationService validationService,
+            SessionService sessionService) {
```
Comment (dbes-gds) [inline account-management-api/src/main/java/uk/gov/di/accountmanagement/lambda/RemoveAccountHandler.java:30]: couple of unused params

File: account-management-api/src/main/java/uk/gov/di/accountmanagement/lambda/RemoveAccountHandler.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                new DynamoService(
+                        configurationService.getAwsRegion(),
+                        configurationService.getEnvironment(),
+                        configurationService.getDynamoEndpointUri());
+    }
+
+    @Override
+    public APIGatewayProxyResponseEvent handleRequest(
+            APIGatewayProxyRequestEvent input, Context context) {
+        try {
```
Comment (dbes-gds) [inline account-management-api/src/main/java/uk/gov/di/accountmanagement/lambda/RemoveAccountHandler.java:46]: add some logging to help with debuggin?

File: account-management-api/src/main/java/uk/gov/di/accountmanagement/lambda/RemoveAccountHandler.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            RemoveAccountRequest removeAccountRequest =
+                    objectMapper.readValue(input.getBody(), RemoveAccountRequest.class);
```
Comment (sanisc) [inline account-management-api/src/main/java/uk/gov/di/accountmanagement/lambda/RemoveAccountHandler.java:48]: you want to read the principal id from the request and do some basic validation to make sure the users subjects match up

**Review by sanisc — CHANGES_REQUESTED** [review]

**Review by JHjava — DISMISSED** [review]

File: ci/terraform/account-management/remove-account.tf [hunk 4]
```diff
@@ -0,0 +1,27 @@
+module "remove_account" {
+  source = "../modules/account-management-endpoint-module"
+
+  endpoint_name   = "remove-account"
```
Comment (sanisc) [inline ci/terraform/account-management/remove-account.tf:4]: can we change this to delete-account

**Review by sanisc — CHANGES_REQUESTED** [review]

**Review by sanisc — APPROVED** [review]

**Pull request merged**